### PR TITLE
cpu/samd51: allow to support ADC0 and ADC1 in the same configuration

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

The current ADC support on SAMD51 only allows to configure lines that use either ADC0 or ADC1. It's not possible different lines that use different ADC device. So for example on the pybadge it's not possible to configure board light sensor (uses ADC1) and Bat voltage measure pin (uses ADC0).

This PR extend the ADC line configuration with an extra pointer to the ADC device. This is only possible on cpu with ADC0 and ADC1 available.

The adafruit-pybadge, adafruit-itsybitsy and same52-xpro board configurations are adapted accordingly and on the pybadge, the configuration is extended with more lines that use ADC0 device, including the VMEAS pin.

### Testing procedure

- Green Murdock
- ADC configuration is still working on modified boards. I haven't tried the new ADC configured on the Pybadge, because I'm missing hardware for that but the light sensor is still working as before
- Code size should remain unchange on other CPU. I checked with `tests/periph_adc` on sam21-xpro and this is the case (I can report the output if requested)

### Issues/PRs references

Noticed that limitation when working on #17807

### CODE REVIEW

File: boards/adafruit-pybadge/include/periph_conf.h [hunk 1]
```diff
@@ -269,13 +269,16 @@ static const sam0_common_usb_config_t sam_usbdev_config[] = {
...
 #define ADC_REF_DEFAULT                     ADC_REFCTRL_REFSEL_INTVCC1
-#define ADC_DEV                             ADC1
 
 static const adc_conf_chan_t adc_channels[] = {
-    /* port, pin, muxpos */
-    {GPIO_PIN(PB, 8), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN2)}, /* A2 */
-    {GPIO_PIN(PB, 9), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN3)}, /* A3 */
-    {GPIO_PIN(PB, 4), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN6)}, /* A7 - Light sensor */
+    /* port, pin, muxpos, dev */
+    {GPIO_PIN(PA, 5), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN5), ADC0},   /* A1 */
```
Comment (dylad) [inline boards/adafruit-pybadge/include/periph_conf.h:275]: Any reason not to add A0 ?

File: boards/adafruit-pybadge/include/periph_conf.h [hunk 2]
```diff
@@ -269,13 +269,16 @@ static const sam0_common_usb_config_t sam_usbdev_config[] = {
...
+    {GPIO_PIN(PB, 8), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN2), ADC1},   /* A2 */
+    {GPIO_PIN(PB, 9), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN3), ADC1},   /* A3 */
```
Comment (dylad) [inline boards/adafruit-pybadge/include/periph_conf.h:277]: PB09 -> ADC0, AIN3 / ADC1,AIN1

**Review by dylad — CHANGES_REQUESTED** [review]
Spotted a few misconfigurations.

File: boards/adafruit-pybadge/include/periph_conf.h (hunk 1 above)
Comment (aabadie) [reply boards/adafruit-pybadge/include/periph_conf.h:275]: It's used by the speaker

File: boards/adafruit-pybadge/include/periph_conf.h [hunk 3]
```diff
@@ -269,13 +269,16 @@ static const sam0_common_usb_config_t sam_usbdev_config[] = {
...
-#define ADC_DEV                             ADC1
 
 static const adc_conf_chan_t adc_channels[] = {
-    /* port, pin, muxpos */
-    {GPIO_PIN(PB, 8), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN2)}, /* A2 */
-    {GPIO_PIN(PB, 9), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN3)}, /* A3 */
-    {GPIO_PIN(PB, 4), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN6)}, /* A7 - Light sensor */
+    /* port, pin, muxpos, dev */
+    {GPIO_PIN(PA, 5), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN5), ADC0},   /* A1 */
+    {GPIO_PIN(PB, 8), ADC_INPUTCTRL_MUXPOS(ADC_INPUTCTRL_MUXPOS_AIN2), ADC1},   /* A2 */
```
[the remark this answers is not in the document]
Comment (aabadie) [reply boards/adafruit-pybadge/include/periph_conf.h:276]: Thanks, good spot. I was confused by the Adafruit schematics and haven't checked the datasheet.

**Comment by aabadie** [comment]
Tested all ADC lines configured and everything is fine now. Thanks!

**Review by dylad — APPROVED** [review]
Looks good to me.

**Pull request merged**